pisek/visualize.py

```diff
@@ -13,304 +9,330 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
-
-from collections import namedtuple
-import fnmatch
+from dataclasses import dataclass
 import json
-from math import ceil
+from math import ceil, inf
 import os
-import re
-import sys
-from colorama import Fore
-from typing import Optional, Union, Iterable, Callable
-
-from pisek.env.task_config import load_config, TaskConfig, SubtaskConfig
-from pisek.jobs.parts.task_job import TaskHelper
-
-VERDICTS = {
-    "ok": "·",
-    "partial": "P",
-    "timeout": "T",
-    "timeout_limited": "t",
-    "wrong_answer": "W",
-    "error": "!",
-}
-VERDICTS_ORDER = ["·", "P", "t", "T", "W", "!"]
-
-# subtask section
-TestCaseResult = namedtuple("TestCaseResult", ("name", "verdict", "value", "points"))
-
-
-def red(msg: str) -> str:
-    return f"{Fore.RED}{msg}{Fore.RESET}"
-
-
-def group_by_subtask(
-    results: list[TestCaseResult], config: TaskConfig
-) -> dict[int, list[TestCaseResult]]:
-    subtasks: dict[int, list[TestCaseResult]] = {
-        num: [] for num, _ in config.subtasks.items()
-    }
-    for result in results:
-        for i, subtask in config.subtasks.items():
-            if in_subtask(result.name, subtask):
-                subtasks[i].append(result)
-    return subtasks
-
+from typing import Optional
+
+from pisek.utils.text import pad, tab, colored
+from pisek.utils.terminal import terminal_width
+from pisek.jobs.parts.solution_result import Verdict
+from pisek.env.task_config import load_config, TaskConfig
+
+test_name_max_length = 0
```

Comment:
Is a global variable really necessary?
Couldn't you calculate the maximum length in `visualize` and pass it as a parameter to `to_str`?